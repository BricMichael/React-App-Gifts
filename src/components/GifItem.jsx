import React from 'react';
import PropTypes from 'prop-types';

const GifItem = ({ url, title }) => {

    return(
        <div className = "card animate__animated animate__bounceIn">

            <img src={ url } alt={ title } className ='imgs'/>
            <p>{title}</p>
        </div>
    );
} 

GifItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired
}

export default GifItem;