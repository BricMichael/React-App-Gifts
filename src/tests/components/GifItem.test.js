import React from 'react';
import { shallow } from "enzyme";
import GifItem from "../../components/GifItem";


describe('Pruebas en el componente ¡¡GigItem!!', () => {

    const title = 'Title image';
    const url = 'https://www.images/gifs/limit:10/.com';

    const wrapper = shallow(<GifItem  title = { title } url={ url } />);
    const img = wrapper.find('img'); //reutilizacion de variable

    test('Capture del jsx componente GifItem', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('Asegurarse que la etiqueta <p> este usando el props title', () => {
        const p = wrapper.find('p').text()
        
        expect( p ).toBe( title )

    });

    test('Asegurarse de la src y alt propiedades, de la etiqueta <img> este usando los  props', () => {
        const src = img.prop('src');
        const alt = img.prop('alt');

        expect( src ).toBe( url );
        expect( alt ).toBe( title )
       
    });

    test('La <img> debe tener la className = animate__bounceIn', () => {
        const div = wrapper.find('div').prop('className').includes('animate__bounceIn');
        const bool = div ? 'Yes' : 'No';
        
        expect( bool ).toBe( 'Yes' ); 

    });

})