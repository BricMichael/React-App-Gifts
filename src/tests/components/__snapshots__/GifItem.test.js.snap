// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente ¡¡GigItem!! Capture del jsx componente GifItem 1`] = `
<div
  className="card animate__animated animate__bounceIn"
>
  <img
    alt="Title image"
    className="imgs"
    src="https://www.images/gifs/limit:10/.com"
  />
  <p>
    Title image
  </p>
</div>
`;
